src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt

```diff
@@ -198,6 +198,11 @@ fun processPathResolveVariants(lookup: ImplLookup, path: RsPath, isCompletion: B
     }
 
     if (qualifier != null) {
+        val primitiveType = TyPrimitive.fromPath(qualifier);
+        if (primitiveType != null) {
+            if (processAssociatedFunctionsAndMethodsDeclarations(lookup, primitiveType, processor)) return true
```

Comment:
Not sure if processAssociatedFunctionsAndMethodsDeclarations is the right thing to call here but it seems to work nicely.

Reply:
Turns out that processing associated items is a mess right now, see #1683. I hope I'll fix it today.